nodejs/eks/nodes/nodegroup.ts

```diff
@@ -586,22 +584,22 @@ function createNodeGroup(
     parent: pulumi.ComponentResource,
     provider?: pulumi.ProviderResource,
 ): NodeGroupData {
-    const instanceProfileName = core.apply((c) =>
-        resolveInstanceProfileName(name, args, c, parent),
-    );
+    const validationErrors: pulumi.InputPropertyErrorDetails[] = [];
+
+    const instanceProfileName = core.apply((c) => resolveInstanceProfileName(args, c));
 
     if (args.clusterIngressRule && args.clusterIngressRuleId) {
-        throw new pulumi.ResourceError(
-            `invalid args for node group ${name}, clusterIngressRule and clusterIngressRuleId are mutually exclusive`,
-            parent,
-        );
+        validationErrors.push({
+            propertyPath: "clusterIngressRule",
+            reason: `clusterIngressRule and clusterIngressRuleId are mutually exclusive`,
+        });
     }
 
     if (args.nodeSecurityGroup && args.nodeSecurityGroupId) {
-        throw new pulumi.ResourceError(
-            `invalid args for node group ${name}, nodeSecurityGroup and nodeSecurityGroupId are mutually exclusive`,
-            parent,
-        );
+        validationErrors.push({
```

Comment:
If we no longer throw here and keep going, can that lead to us being in a bad place? When we use validation errors like this, do we need to make sure we throw validation errors _before_ _any_ resources are created?

In this case I think we do throw before creating resources, but I'm just curious for future reference.